core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -405,9 +405,7 @@ class SparkContext(config: SparkConf) extends Logging {
 
     _eventLogDir =
       if (isEventLogEnabled) {
-        val unresolvedDir = conf.get("spark.eventLog.dir", EventLoggingListener.DEFAULT_LOG_DIR)
-          .stripSuffix("/")
-        Some(Utils.resolveURI(unresolvedDir))
```

Comment:
Shouldn't you URI encode this `unresolvedDir` here? I think encode space to "%20" should be enough. Not sure why you need to change lots of code.

```
scala> val a = new URI("/tmp/aa%20nn")
a: java.net.URI = /tmp/aa%20nn

scala> val path = new Path(a)
path: org.apache.hadoop.fs.Path = /tmp/aa nn
```